### ISSUE TITLE
🐛 Bug Report — Runtime APIs: Error: [unenv] crypto.X509Certificate is not implemented yet!

### ISSUE THREAD

**Issue by 1yasa** [cloudflare/workerd#2720]
Codes:
<img width="619" alt="截屏2024-09-17 20 47 22" src="https://github.com/user-attachments/assets/65831010-9661-4cb6-8f9d-e1afe32231d0">

Error:
<img width="1100" alt="截屏2024-09-17 20 47 00" src="https://github.com/user-attachments/assets/da26a56a-3699-4108-8bb3-ef384bd623e2">

Crypto supports on workers doc:
<img width="646" alt="截屏2024-09-17 20 48 44" src="https://github.com/user-attachments/assets/488b4c04-62e1-4dcb-9c10-c3ec3c499eff">

Config:
<img width="447" alt="截屏2024-09-17 20 50 08" src="https://github.com/user-attachments/assets/0a42b79e-2a0d-49e1-983d-c0b27702a76e">

So my question is, why does the official website state that it supports crypto.X509Certificate, and in the code, it can also print out that X509Certificate is an anonymous class, why does it report an error when used: `Error: [unenv] crypto.X509Certificate is not implemented yet!`

Both local and deploy to the server.

I really need help with this; if Cloudflare can't use the X509Certificate, it will be very troublesome!

**Comment by 1yasa** [comment]
Success with `compatibility_flags = ["nodejs_compat"]`:
<img width="435" alt="image" src="https://github.com/user-attachments/assets/dce8f541-cba9-4b91-9a7d-6dec51b1f7b9">

But `compatibility_flags = ["nodejs_compat_v2"]` can treat `cryptp` as `node:crypto`, even `import * from 'crypto'` in packages, it use unenv as polyfills for nodejs apis,  However, when it polyfills the crypto module for Node.js, it ignores the internal implementation of workerd. 

So is there a way to merge the Node.js API implementation of workerd with that of unenv (using workerd's implementation when the API is not implemented in unenv, instead of throwing an Error: [unenv] crypto.X509Certificate is not implemented yet!)?

**Comment by IgorMinar** [comment]
it's because this code is outdated now that we have a native crypto implementation:

https://github.com/unjs/unenv/blob/main/src/runtime/node/crypto/%24cloudflare.ts#L12-L41

related issue: https://github.com/cloudflare/workerd/issues/2097

more recent discussion on how to stay up to date: https://github.com/unjs/unenv/issues/294

**Comment by anonrig** [comment]
This will be fixed with the next release.

**Comment by 1yasa** [comment]
> This will be fixed with the next release.

<img width="753" alt="image" src="https://github.com/user-attachments/assets/56d4e97c-4c25-46fc-bb25-7f7a7bad19e9">

`Unrecognized or unimplemented EC curve \"id-ecPublicKey\" requested` when use crypto with `compatibility_flags = ["nodejs_compat"]`.

Here is the error source:

https://github.com/agisboye/app-store-server-api/blob/main/src/Decoding.ts#L47

<img width="722" alt="截屏2024-09-19 21 16 03" src="https://github.com/user-attachments/assets/29d51933-3ed8-451c-ac42-1d96a212fd1d">

Here is the error detail:
<img width="1049" alt="截屏2024-09-19 21 17 04" src="https://github.com/user-attachments/assets/68e63f9f-6528-41d8-b2f9-7d59baef6042">

So my question is, the X509 implemented by workerd seems to be completely unusable when using the decodeJWS function of /app-store-server-api.

Did I make a mistake, or does the X509 implementation in workerd indeed throw an error when trying to get the public key with cert.publicKey? Below is my business code (for reproduction):

```ts
const app_store_api = getAppStoreApi()
const res_subscription = await app_store_api.getSubscriptionStatuses(tid!)

const item = res_subscription.data[0].lastTransactions.find(item => item.originalTransactionId === tid)!

const res_transaction = await decodeTransaction(item.signedTransactionInfo)
```

**Comment by vicb** [comment]
@1yasa the PR referenced above on cloudflare/workers-sdk will expose the workerd implentation for `crypto.X509Certificate` when the `nodejs_compat_v2` flag is used.

This should be released in wrangler 3.78.7 later today. Please note that you can test now by using the wrangler build in https://github.com/cloudflare/workers-sdk/pull/6782#issuecomment-2362936857

Once this is released we will close the issue "Runtime APIs: Error: [unenv] crypto.X509Certificate is not implemented yet!".

If you still see problems, please open a distinct issue to make tracking easier.

Thanks!

**Comment by 1yasa** [comment]
> @1yasa the PR referenced above on cloudflare/workers-sdk will expose the workerd implentation for `crypto.X509Certificate` when the `nodejs_compat_v2` flag is used.
> 
> This should be released in wrangler 3.78.7 later today. Please note that you can test now by using the wrangler build in [cloudflare/workers-sdk#6782 (comment)](https://github.com/cloudflare/workers-sdk/pull/6782#issuecomment-2362936857)
> 
> Once this is released we will close the issue "Runtime APIs: Error: [unenv] crypto.X509Certificate is not implemented yet!".
> 
> If you still see problems, please open a distinct issue to make tracking easier.
> 
> Thanks!

Get. Thanks!
